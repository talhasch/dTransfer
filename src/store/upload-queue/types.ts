

export enum ActionTypes {
    SET = '@files/SET',
    RESET = '@files/RESET',
    START = '@files/START',
    FINISH = '@files/FINISH',
    ITEM_START = '@files/ITEM_START',
    ITEM_PROGRESS = '@files/ITEM_PROGRESS',
    ITEM_FINISH = '@files/ITEM_FINISH',
    ITEM_ERROR = '@files/ITEM_ERROR',
}

interface BaseAction {
    type: ActionTypes
}

export interface SetAction extends BaseAction {
    type: ActionTypes.SET;
    files: Array<File>;
}

export interface ResetAction extends BaseAction {
    type: ActionTypes.RESET;
}

export interface StartAction extends BaseAction {
    type: ActionTypes.START
}

export interface FinishAction extends BaseAction {
    type: ActionTypes.FINISH
}

export interface ItemStartAction extends BaseAction {
    type: ActionTypes.ITEM_START,
    itemId: number
}

export interface ItemProgressAction extends BaseAction {
    type: ActionTypes.ITEM_PROGRESS,
    itemId: number
}

export interface ItemFinishAction extends BaseAction {
    type: ActionTypes.ITEM_FINISH,
    itemId: number
}

export interface ItemErrorAction extends BaseAction {
    type: ActionTypes.ITEM_ERROR,
    itemId: number
}

export type Action = SetAction | ResetAction | ResetAction | StartAction | FinishAction | ItemStartAction | ItemProgressAction | ItemFinishAction | ItemErrorAction;


export interface Item {
    id: number,
    obj: File,
    status: ItemStatus,
    progress: number
}

export enum ItemStatus {
    READY = 1,
    UPLOADING = 2,
    UPLOADED = 3
}


export interface State {
    list: Array<Item>,
    inProgress: boolean
}

export const initialState: State = {
    list: [],
    inProgress: false
};
