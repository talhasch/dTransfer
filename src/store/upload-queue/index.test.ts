import reducer, {
    addAct,
    itemDeleteAct,
    startAct,
    finishAct,
    itemStartAct,
    itemProgressAct,
    itemFinishAct
} from "./index";

import {State, initialState} from './types';

let state: State = initialState;

it('1- default state', () => {
    expect(state).toMatchSnapshot();
});

it('2- add to queue', () => {
    const files = [
        new File([new ArrayBuffer(10)], 'document.pdf', {lastModified: 1589567882559}),
        new File([new ArrayBuffer(10)], 'sheet.xls', {lastModified: 1589567882559})
    ];

    state = reducer(state, addAct(files));
    expect(state).toMatchSnapshot();
});


it('3- add to queue', () => {
    const files = [
        new File([new ArrayBuffer(10)], 'document.pdf', {lastModified: 1589567882559}),
        new File([new ArrayBuffer(10)], 'sheet.xls', {lastModified: 1589567882559}),
        new File([new ArrayBuffer(10)], 'image.jpg', {lastModified: 1589567882559})
    ];

    state = reducer(state, addAct(files));
    expect(state).toMatchSnapshot();
});


it('4- delete item from queue', () => {
    state = reducer(state, itemDeleteAct('sheet.xls-10-1589567882559'));
    expect(state).toMatchSnapshot();
});

it('5- start', () => {
    state = reducer(state, startAct());
    expect(state).toMatchSnapshot();
});

it('6- item start', () => {
    state = reducer(state, itemStartAct('image.jpg-10-1589567882559'));
    expect(state).toMatchSnapshot();
});

it('7- item progress', () => {
    state = reducer(state, itemProgressAct('image.jpg-10-1589567882559', 11));
    expect(state).toMatchSnapshot();
});

it('8- item finish', () => {
    state = reducer(state, itemFinishAct('image.jpg-10-1589567882559'));
    expect(state).toMatchSnapshot();
});



it('52- finish', () => {
    state = reducer(state, finishAct());
    expect(state).toMatchSnapshot();
});
