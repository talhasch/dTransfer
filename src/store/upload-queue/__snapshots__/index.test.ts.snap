// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `
Object {
  "inProgress": false,
  "list": Array [],
}
`;

exports[`2- add to queue 1`] = `
Object {
  "inProgress": false,
  "list": Array [
    Object {
      "error": "",
      "id": "document.pdf-10-1589567882559",
      "obj": File {},
      "progress": 0,
      "status": 1,
      "url": "",
    },
    Object {
      "error": "",
      "id": "sheet.xls-10-1589567882559",
      "obj": File {},
      "progress": 0,
      "status": 1,
      "url": "",
    },
  ],
}
`;

exports[`3- add to queue 1`] = `
Object {
  "inProgress": false,
  "list": Array [
    Object {
      "error": "",
      "id": "image.jpg-10-1589567882559",
      "obj": File {},
      "progress": 0,
      "status": 1,
      "url": "",
    },
    Object {
      "error": "",
      "id": "document.pdf-10-1589567882559",
      "obj": File {},
      "progress": 0,
      "status": 1,
      "url": "",
    },
    Object {
      "error": "",
      "id": "sheet.xls-10-1589567882559",
      "obj": File {},
      "progress": 0,
      "status": 1,
      "url": "",
    },
  ],
}
`;

exports[`4- delete item from queue 1`] = `
Object {
  "inProgress": false,
  "list": Array [
    Object {
      "error": "",
      "id": "image.jpg-10-1589567882559",
      "obj": File {},
      "progress": 0,
      "status": 1,
      "url": "",
    },
    Object {
      "error": "",
      "id": "document.pdf-10-1589567882559",
      "obj": File {},
      "progress": 0,
      "status": 1,
      "url": "",
    },
  ],
}
`;
